playground/frontend/playground_components/lib/src/widgets/run_button.dart

```diff
@@ -73,57 +72,64 @@ class RunButton extends StatelessWidget {
                     )
                   : const Icon(Icons.play_arrow),
               label: isRunning
-                  ? PeriodicRebuilderWidget(
-                      rebuildFrequencyMls: 1,
+                  ? PeriodicBuilderWidget(
+                      interval: const Duration(
+                        milliseconds: _buttonTextRebuildInterval,
+                      ),
                       builder: () {
                         return _ButtonText(
                           isRunning: isRunning,
                           runStartDate: runStartDate,
+                          runStopDate: runStopDate,
                         );
                       },
                     )
                   : _ButtonText(
                       isRunning: isRunning,
                       runStartDate: runStartDate,
+                      runStopDate: runStopDate,
                     ),
-              onPressed: _onPressHandler(),
+              onPressed: isEnabled ? _onPressed() : null,
             ),
           ),
         );
       },
     );
   }
 
-  VoidCallback? _onPressHandler() {
-    if (disabled) {
-      return null;
-    }
-    return !playgroundController.codeRunner.isCodeRunning ? runCode : cancelRun;
+  VoidCallback _onPressed() {
+    return playgroundController.codeRunner.isCodeRunning ? cancelRun : runCode;
   }
 }
 
 class _ButtonText extends StatelessWidget {
   final bool isRunning;
   final DateTime? runStartDate;
+  final DateTime? runStopDate;
 
   const _ButtonText({
     required this.isRunning,
     required this.runStartDate,
+    required this.runStopDate,
   });
 
+  static const _msToSec = 1000;
+  static const _secondsFractions = 1;
```

Comment:
```suggestion
  static const _secondsPrecision = 1;
```

This is also related to `_buttonTextRebuildInterval` and should likely be derived from it.

Otherwise if we change the interval to 1000 ms, we would have weird jumps: 1.1 -> 2.1 -> 3.1...

Reply:
I haven't noticed weird jumps, but when `_buttonTextRebuildInterval` is changed to 1000 mls, it is noticeable that `PeriodicBuilderWidget` is also rebuilt when `CodeRunner` notifies listeners.